src/eos-updater-fetch.c

```diff
@@ -295,18 +299,460 @@ content_fetch_old (EosUpdater      *updater,
   return TRUE;
 }
 
-static void
-content_fetch (GTask *task,
-               gpointer object,
-               gpointer task_data,
-               GCancellable *cancellable)
+
+static GFile *
+get_temporary_directory_to_checkout_in (GError **error)
 {
-  EosUpdater *updater = EOS_UPDATER (object);
-  EosUpdaterData *data = task_data;
-  g_autoptr(GError) error = NULL;
-  g_autoptr(GMainContext) task_context = g_main_context_new ();
+  g_autofree gchar *temp_dir = g_dir_make_tmp ("ostree-checkout-XXXXXX", error);
+  g_autofree gchar *path = NULL;
 
-  g_main_context_push_thread_default (task_context);
+  if (!temp_dir)
+    return NULL;
+
+  path = g_build_filename (temp_dir, "checkout", NULL);
+  return g_file_new_for_path (path);
+}
+
+static GFile *
+inspect_directory_in_ostree_repo (OstreeRepo    *repo,
+                                  const gchar   *checksum,
+                                  const gchar   *subpath,
+                                  GCancellable  *cancellable,
+                                  GError       **error)
+{
+  g_autofree gchar *checkout_directory_path = NULL;
+  g_autoptr(GFile) checkout_directory = NULL;
+  OstreeRepoCheckoutAtOptions options = { 0, };
+
+  checkout_directory = get_temporary_directory_to_checkout_in (error);
+
+  if (!checkout_directory)
+    return NULL;
+
+  checkout_directory_path = g_file_get_path (checkout_directory);
+
+  /* Now that we have a temporary directory, checkout the OSTree in it
+   * at the nominated path */
+  options.subpath = subpath;
+
+  if (!ostree_repo_checkout_at (repo,
+                                &options,
+                                -1,
+                                checkout_directory_path,
+                                checksum,
+                                cancellable,
+                                error))
+    {
+      eos_updater_remove_recursive (checkout_directory, NULL);
+      return NULL;
+    }
+
+  return g_steal_pointer (&checkout_directory);
+}
+
+static GHashTable *
+flatpak_ref_actions_for_commit (OstreeRepo    *repo,
```

Comment:
Can this be moved out to the utility library please? Then it can be easily tested on a temporary `OstreeRepo` without having to queue up `eos-updater` with a fetch operation for that repo.